tests/api_tests/collections/test_views.py

```diff
@@ -0,0 +1,716 @@
+        self.non_contrib.set_password('justapoorboy')
+        self.non_contrib.save()
+        self.basic_non_contrib_auth = (self.non_contrib.username, 'justapoorboy')
+
+        self.deleted = ProjectFactory(is_deleted=True, is_folder=True)
+        self.private = ProjectFactory(is_public=False, creator=self.user, is_folder=True)
+        self.public = ProjectFactory(is_public=True, creator=self.user, is_folder=True)
+        self.url = '/v2/collections/'
+
+    def test_only_returns_non_deleted_public_projects(self):
+        res = self.app.get(self.url)
+        node_json = res.json['data']
+
+        ids = [each['id'] for each in node_json]
+        assert_in(self.public._id, ids)
+        assert_not_in(self.deleted._id, ids)
+        assert_not_in(self.private._id, ids)
+
+    def test_return_public_collection_list_logged_out_user(self):
+        res = self.app.get(self.url)
+        assert_equal(res.status_code, 200)
+        ids = [each['id'] for each in res.json['data']]
+        assert_in(self.public._id, ids)
+        assert_not_in(self.private._id, ids)
+
+    def test_return_public_collection_list_logged_in_user(self):
+        res = self.app.get(self.url, auth=self.non_contrib)
+        assert_equal(res.status_code, 200)
+        ids = [each['id'] for each in res.json['data']]
+        assert_in(self.public._id, ids)
+        assert_not_in(self.private._id, ids)
+
+    def test_return_private_collection_list_logged_out_user(self):
+        res = self.app.get(self.url)
+        ids = [each['id'] for each in res.json['data']]
+        assert_in(self.public._id, ids)
+        assert_not_in(self.private._id, ids)
+
+        Node.remove()
+
+class TestCollectionFiltering(ApiTestCase):
+    def setUp(self):
+        ApiTestCase.setUp(self)
+        self.user_one = UserFactory.build()
+        self.user_one.set_password('justapoorboy')
+        self.user_one.save()
+        self.basic_auth_one = (self.user_one.username, 'justapoorboy')
+        self.user_two = UserFactory.build()
+        self.user_two.set_password('justapoorboy')
+        self.user_two.save()
+        self.basic_auth_two = (self.user_two.username, 'justapoorboy')
+        self.project_one = ProjectFactory(title="Project One", is_public=True, is_folder=True)
+        self.project_two = ProjectFactory(title="Project Two", is_folder=True, is_public=True)
+        self.project_three = ProjectFactory(title="Three", is_public=True, is_folder=True)
+        self.private_project_user_one = ProjectFactory(title="Private Project User One", is_public=False,
+                                                       is_folder=True, creator=self.user_one)
+        self.private_project_user_two = ProjectFactory(title="Private Project User Two", is_public=False,
+                                                       is_folder=True, creator=self.user_two)
+        self.folder = FolderFactory()
+        self.dashboard = DashboardFactory()
+
+        self.url = "/v2/collections/"
```

Comment:
This will be happening independently for existing tests, but see #3062 to make this more futureproof.